### PR TITLE
Text analytics update readme, add tests

### PR DESCRIPTION
**Pull request by samvaity**

### CODE REVIEW

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java [hunk 1]
```diff
@@ -19,11 +19,13 @@
 
     @Override
     protected void beforeTest() {
-        beforeTestSetup();
-        client = clientSetup(httpPipeline -> new TextAnalyticsClientBuilder()
-            .endpoint(getEndPoint())
-            .pipeline(httpPipeline)
-            .buildAsyncClient());
+//        beforeTestSetup();
```
Comment (mssfang) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java]: Can you run a test if you comment out these lines?

File: sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java [hunk 2]
```diff
@@ -97,7 +97,6 @@ public TextAnalyticsServiceVersion getServiceVersion() {
     }
 
     Mono<Response<DetectLanguageResult>> detectLanguageWithResponse(String text, String countryHint, Context context) {
-        Objects.requireNonNull(text, "'text' cannot be null.");
```
Comment (mssfang) [inline sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java:100]: what reason to remove requireNonNull check for `text` input

Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java:100]: On the client-side we should be doing as fewer validations as possible as service side is already doing those.

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java (hunk 1 above)
Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java]: My IDE isn't picking up the env variables correctly so for now doing it by using the subscription key manually.

Comment (samvaity) [reply sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java]: @mssfang  could you run it in live mode and transfer the test records on the main PR, till my IDE starts to pick up the vars

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java [hunk 3]
```diff
@@ -21,9 +23,9 @@
     protected void beforeTest() {
         beforeTestSetup();
         client = clientSetup(httpPipeline -> new TextAnalyticsClientBuilder()
-            .endpoint(getEndPoint())
-            .pipeline(httpPipeline)
-            .buildAsyncClient());
+                                                 .endpoint(getEndPoint())
```
Comment (mssfang) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java:26]: extra spaces

File: sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java [hunk 4]
```diff
@@ -121,24 +120,25 @@ public void detectLanguageEmptyText() {
...
-
     /**
      * Verifies that detectLanguage returns an "UNKNOWN" result when faulty text is passed.
      */
     @Test
     public void detectLanguageFaultyText() {
         StepVerifier.create(client.detectLanguage("!@#%%"))
             .assertNext(response ->
-                assertEquals(response.getPrimaryLanguage().getIso6391Name(), "(Unknown)"))
+                            assertEquals(response.getPrimaryLanguage().getIso6391Name(), "(Unknown)"))
```
Comment (mssfang) [inline sdk/textanalytics/azure-ai-textanalytics/src/test/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClientTest.java:122]: extra space

**Review by mssfang — APPROVED** [review]

**Pull request merged**